### DISCUSSION TITLE
What's the simplest Python way to render a CSV as markdown table?

### DISCUSSION THREAD

**Discussion by aborruso** [quarto-dev/quarto-cli#2859] [Q&A] [answered]
Hi,
I need to render a markdown table programmatically, starting from a CSV file.

It's not

~~~
```{python}
import pandas as pd

df = pd.read_csv("your_csv_file.csv")
df.to_markdown()
```
~~~

I would like to have 1) and not 2).

Thank you

![image](https://user-images.githubusercontent.com/30607/195901192-9c968842-5842-41d3-b749-60601eb8c78d.png)

**Comment by cscheid** [comment, marked as answer]
You need to tell IPython that the output is supposed to be interpreted as Markdown, instead of a string.

Try:

```python
from IPython.display import display, Markdown
display(Markdown(df.to_markdown()))
```

**Reply by aborruso** [reply]
I had solved also using

```
import pandas as pd
from tabulate import tabulate
import IPython.display as d


df = pd.read_csv("your_csv_file.csv")
md = tabulate(df, headers='keys', tablefmt='pipe',showindex=False)
d.Markdown(md)
```

**Reply by clabornd** [reply]
For me, using the knitr engine to write a pdf, this works:

`Markdown(df.to_markdown())`

this:
`display(Markdown(df.to_markdown()))`

outputs `<IPython.core.display.Markdown object>`
